src/Services/EncounterService.php

```diff
@@ -242,6 +242,54 @@ public function getEncounterForPatient($pid, $eid)
         return sqlQuery($sql, array($pid, $eid));
     }
 
+    public function insertEncounter($pid, $data)
+    {
+
```

Comment:
when you add an encounter, the code will look something like this (copy/pasted this from the interface/forms/newpatient/save.php script):
```
    $encounter = generate_id();
    addForm(
        $encounter,
        "New Patient Encounter",
        sqlInsert(
            "INSERT INTO form_encounter SET
                date = ?,      
                onset_date = ?,
                reason = ?,
                facility = ?,
                pc_catid = ?,
                facility_id = ?,
                billing_facility = ?,
                sensitivity = ?,
                referral_source = ?,
                pid = ?,
                encounter = ?,
                pos_code = ?,
                external_id = ?,
                provider_id = ?,
                parent_encounter_id = ?",
            [
                $date,
                $onset_date,
                $reason,
                $facility,
                $pc_catid,
                $facility_id,
                $billing_facility,
                $sensitivity,
                $referral_source,
                $pid,
                $encounter,
                $pos_code,
                $external_id,
                $provider_id,
                $parent_enc_id
            ]
        ),
        "newpatient",
        $pid,
        $userauthorized,
        $date
    );
```

Reply:
this means will need a require_once at top of this script for library/forms.inc script (someday, it will be in a class, but for now gotta go this route to be able to use the addForm function)

Reply:
btw, the $userauthorized is same as $data["provider_id"]

Reply:
Following above strategy will mean that the encounter is created just like it is done natively in OpenEMR

Reply:
What does the Addform do?

Reply:
https://github.com/openemr/openemr/blob/master/library/forms.inc#L44-L84
(when creating an encounter, an entry in forms and form_encounter is created)